### PR TITLE
br: Update mysql.stats_meta after checksum in restore

### PR DESCRIPTION
**Pull request by joccau**
Signed-off-by: joccau <zak.zhao@pingcap.com>

### What problem does this PR solve?

Issue Number: close https://github.com/pingcap/tidb/issues/29216

Problem Summary:
The mysql.stats_meta miss their updates and the row_count is still 0 after restore

### Check List

Tests 

- [x] Manual test (add detailed scripts or steps below)

After restore, show the stats_meta
image

### Release note

```release-note
Update mysql.stats_meta after checksum in restore
```

### CODE REVIEW

File: br/pkg/restore/client.go [hunk 1]
```diff
@@ -906,19 +911,35 @@ func (rc *Client) statLoader(ctx context.Context, input <-chan *CreatedTable) {
 			if !ok {
 				return
 			}
+
+			// Not need to return err when failed because of update analysis-meta
+			if tbl.OldTable.Crc64Xor > 0 {
```
Comment (kennytm) [inline br/pkg/restore/client.go:916]: `Crc64Xor = 0` did not mean the table was empty.

File: br/pkg/restore/db.go [hunk 2]
```diff
@@ -92,6 +93,31 @@ func (db *DB) ExecDDL(ctx context.Context, ddlJob *model.Job) error {
...
+
+	// set restoreTS to snapshot ans version
+	updateMetaSQL := fmt.Sprintf(
+		"update %s.%s set snapshot = %d, version = %d, count = %d where table_id = %d",
+		sysDB,
+		statsMetaTbl,
+		restoreTS,
+		restoreTS,
+		count,
+		tableID,
```
Comment (kennytm) [inline br/pkg/restore/db.go:109]: use prepared SQL for these 4

**Review by winoros — APPROVED** [review]

**Review by kennytm — APPROVED** [review]

**Pull request merged**